azurerm/resource_arm_storage_management_policy.go

```diff
@@ -0,0 +1,389 @@
+	if len(ref) == 0 {
+		return nil, fmt.Errorf("Error: storage management policy rule should be defined")
+	}
+
+	name := ref["name"].(string)
+	enabled := ref["enabled"].(bool)
+	typeVal := ref["type"].(string)
+
+	definition := storage.ManagementPolicyDefinition{
+		Filters: &storage.ManagementPolicyFilter{},
+		Actions: &storage.ManagementPolicyAction{},
+	}
+	filtersRef := ref["filters"].([]interface{})
+	if len(filtersRef) == 1 {
+		filterRef := filtersRef[0].(map[string]interface{})
+
+		prefixMatches := []string{}
+		prefixMatchesRef := filterRef["prefix_match"].(*schema.Set)
+		if prefixMatchesRef != nil {
+			for _, prefixMatchRef := range prefixMatchesRef.List() {
+				prefixMatches = append(prefixMatches, prefixMatchRef.(string))
+			}
+		}
+		definition.Filters.PrefixMatch = &prefixMatches
+
+		blobTypes := []string{}
+		blobTypesRef := filterRef["blob_types"].(*schema.Set)
+		if blobTypesRef != nil {
+			for _, blobTypeRef := range blobTypesRef.List() {
+				blobTypes = append(blobTypes, blobTypeRef.(string))
+			}
+		}
+		definition.Filters.BlobTypes = &blobTypes
+	}
+	actionsRef := ref["actions"].([]interface{})
+	if len(actionsRef) == 1 {
+		actionRef := actionsRef[0].(map[string]interface{})
+
+		baseBlobsRef := actionRef["base_blob"].([]interface{})
+		if len(baseBlobsRef) == 1 {
+			baseBlob := &storage.ManagementPolicyBaseBlob{}
+			baseBlobRef := baseBlobsRef[0].(map[string]interface{})
+			if v, ok := baseBlobRef["tier_to_cool_after_days_since_modification_greater_than"]; ok {
+				v2 := float64(v.(int))
+				baseBlob.TierToCool = &storage.DateAfterModification{DaysAfterModificationGreaterThan: &v2}
+			}
+			if v, ok := baseBlobRef["tier_to_archive_after_days_since_modification_greater_than"]; ok {
+				v2 := float64(v.(int))
+				baseBlob.TierToArchive = &storage.DateAfterModification{DaysAfterModificationGreaterThan: &v2}
+			}
+			if v, ok := baseBlobRef["delete_after_days_since_modification_greater_than"]; ok {
+				v2 := float64(v.(int))
+				baseBlob.Delete = &storage.DateAfterModification{DaysAfterModificationGreaterThan: &v2}
+			}
+			definition.Actions.BaseBlob = baseBlob
+		}
+
+		snapshotRef := actionRef["snapshot"].([]interface{})
+		if len(snapshotRef) == 1 {
+			snapshot := &storage.ManagementPolicySnapShot{}
+			snapshotRef := snapshotRef[0].(map[string]interface{})
+			if v, ok := snapshotRef["delete_after_days_since_creation_greater_than"]; ok {
+				v2 := float64(v.(int))
+				snapshot.Delete = &storage.DateAfterCreation{DaysAfterCreationGreaterThan: &v2}
+			}
+			definition.Actions.Snapshot = snapshot
+		}
+	}
+
+	rule := &storage.ManagementPolicyRule{
+		Name:       &name,
+		Enabled:    &enabled,
+		Type:       &typeVal,
+		Definition: &definition,
+	}
+	return rule, nil
+}
+
+func flattenStorageManagementPolicyRules(armRules *[]storage.ManagementPolicyRule) []interface{} {
+	rules := make([]interface{}, 0)
+
```

Comment:
We need to do another nil check here 
```suggestion
if armRules == nil {
  return rules
}
```